includes/data-stores/class-wc-admin-reports-customers-data-store.php

```diff
@@ -460,56 +460,60 @@ public static function get_existing_customer_id_from_order( $order ) {
 	 * @return int|bool
 	 */
 	public static function get_or_create_customer_from_order( $order ) {
-		global $wpdb;
-		$returning_customer_id = self::get_existing_customer_id_from_order( $order );
+		if( false !== $order){ // Only try if the order details exist
```

Comment:
Rather than wrap this entire block in an `if()` statement, let's just check for the positive case `if ( false === $order )` and `return false;` earlier.

Also, this line doesn't adhere to the WP coding standards. If you `composer install` before committing, PHP Code Sniffer will catch things like this.